main.go

```diff
@@ -141,15 +144,27 @@ func ansibleEnable() {
 }
 
 func getAnsibleRepository(runDir string) error {
-	remoteURL := fmt.Sprintf("%s://%s", viper.GetString("http-proto"), viper.GetString("http-url"))
+	httpURL := viper.GetString("http-url")
+	s3Obj := viper.GetString("s3-obj")
 	localCacheFile := fmt.Sprintf("/tmp/%s.tgz", appName)
 
-	err := idempotentFileDownload(remoteURL, localCacheFile, viper.GetString("http-user"), viper.GetString("http-pass"))
-	if err != nil {
-		return errors.Wrap(err, "unable to pull Ansible repo")
+	// Exactly one variable is defined
+	if (httpURL == "") == (s3Obj == "") {
+		return errors.New("exactly one remote resource must be specified. Choose one 'http-url' or 's3-obj'")
+	} else if httpURL != "" {
+		remoteHTTPURL := fmt.Sprintf("%s://%s", viper.GetString("http-proto"), httpURL)
+		err := idempotentFileDownload(remoteHTTPURL, localCacheFile, viper.GetString("http-user"), viper.GetString("http-pass"))
+		if err != nil {
+			return errors.Wrap(err, "unable to pull Ansible repo")
+		}
+	} else if s3Obj != "" {
+	  err := getter.Get(localCacheFile, s3Obj)
```

Comment:
Do we not have the idempotency and MD5sum checks anymore w/ this? I think we still want this. Or is that somehow baked into the `getter.Get` function already?

Reply:
If `getter.Get` handles this, I think it is worth adding a comment explaining that. Sticks out to me like a descrepancy.

Reply:
Unless I'm missing something, we need a little bit of work to get the checksum behavior we want: https://github.com/hashicorp/go-getter#checksumming

Reply:
Unless the URL would be of the form:
`s3.amazonaws.com/tesla-scratch-ansible-tars/dev/latest.tgz?checksum=file:s3.amazonaws.com/tesla-scratch-ansible-tars/dev/latest.md5`?

Reply:
Haha I had a [comment](https://github.com/teslamotors/ansible_puller/pull/14#discussion_r618698089) but I removed it because I decided to defer this problem to the future

The only reason that check-summing works in Artifactory is that it's a feature baked into Artifactory that allows you to get the md5 tag by appending the md5 suffix to any generic artifact. This doesn't exist in S3. To get the MD5 sum in S3 we would first need to download the object via a native golang client, [extract the ETag](https://docs.aws.amazon.com/sdk-for-go/api/service/s3/#GetObjectOutput), and then re-download the object via go-getter with the extracted MD5 sum from the Etag. This works unless the object was uploaded via multi-part upload to S3.

I didn't do this checksumming because of how hacky and fragile it is. We don't really get a lot out of it either. If you can find a way to get the correct MD5 hash then I'm down to do it!

Reply:
I added a comment!

Reply:
What about looking for a `<path>.md5` file? Could be enabled as an option

Reply:
Using an MD5 sum is a pretty important optimization to prevent pulling the whole tar ball every host every 30 minutes, right? That might amount to a non-trivial S3 egress cost (might be worth doing back of envelope for). We can manually make a `<path>.md5` file as part of our publish pipeline. I like Jesse's suggestion of trying the MD5 sum first.

Reply:
Okay I've done this and added tests! Check it out.

Also, what do y'all think about naming the option something other than s3Obj since this is pretty generic now. Any good ideas?